platform/android/export/gradle_export_util.h

```diff
@@ -242,4 +254,190 @@ String _get_application_tag(const Ref<EditorExportPreset> &p_preset, const Strin
+
+	return OK;
+}
+
+Error _deal_with_pack_files(const String &path, const AssetPackInfo &packInfo) {
+	print_line("path: res://" + path);
+	DirAccess *da = DirAccess::open("res://" + path);
+	if (!da) {
+		return ERR_CANT_OPEN;
+	}
+	da->list_dir_begin();
+	while (true) {
+		String file = da->get_next();
+		print_line(file);
+		if (file == "") {
+			break;
+		}
+		if (file == "." || file == "..") {
+			continue;
+		}
+		if (file.ends_with(".pck")) {
+			String dest = "res://android/build/" + packInfo.packName + "/src/main/assets/" + path + file;
+			da->make_dir_recursive(dest.get_base_dir());
+			print_line("destination: " + dest);
+			da->copy(path + file, dest);
+		} else if (da->dir_exists(file)) {
+			Error err = _deal_with_pack_files(path + file + "/", packInfo);
+			if (err != OK) {
+				da->list_dir_end();
+				return err;
+			}
+		}
+	}
+	da->list_dir_end();
+	return OK;
+}
+
+Error _copy_asset_directory_files(const AssetPackInfo &packInfo) {
+	DirAccess *da = DirAccess::create(DirAccess::ACCESS_RESOURCES);
+	String asset_path_from = "res://android/build/assets/" + packInfo.filepath;
+	String asset_path_to = "res://android/build/" + packInfo.packName + "/src/main/assets/" + packInfo.filepath;
+	print_line(asset_path_to);
+	print_line(asset_path_from);
+	da->make_dir_recursive(asset_path_to);
+	//	Error err = da->copy(asset_path_from, asset_path_to);
+	Error err = da->rename(asset_path_from, asset_path_to);
+	if (err != OK) {
+		print_line("could not rename");
+		return err;
+	}
+	print_line("dealing with pack files now");
+	err = _deal_with_pack_files(packInfo.filepath, packInfo);
+
+	return err;
+}
+
+Error _create_asset_pack_folder(const AssetPackInfo &packInfo) {
+	String packName = packInfo.packName;
+	String deliveryType = "install-time"; //default
+	if (packInfo.deliveryMode == INSTALL_TIME) {
+		deliveryType = "install-time";
+	}
+	if (packInfo.deliveryMode == FAST_FOLLOW) {
+		deliveryType = "fast-follow";
+	}
+	if (packInfo.deliveryMode == ON_DEMAND) {
+		deliveryType = "on-demand";
+	}
+	String build_gradle_string = vformat(
+			"apply plugin: 'com.android.asset-pack'\n\n"
+			"assetPack {\n"
+			"    packName = \"%s\"\n"
+			"    dynamicDelivery {\n"
+			"        deliveryType = \"%s\"\n"
+			"    }\n"
+			"}\n",
+			packName, deliveryType);
+	String build_gradle_path = vformat("res://android/build/%s/build.gradle", packName);
+	store_string_at_path(build_gradle_path, build_gradle_string);
+	Error err = _copy_asset_directory_files(packInfo);
+	return err;
+}
+
+Error _read_asset_pack_config_file(const String config_filepath, Vector<AssetPackInfo> &r_asset_pack_info) {
+	//TODO: read asset_pack file line by line and populate r_asset_pack_info
+	FileAccess *f = FileAccess::open(config_filepath, FileAccess::READ);
+	if (f == NULL) {
+		return ERR_DOES_NOT_EXIST;
+	}
+	while (!f->eof_reached()) {
+		AssetPackInfo temp_info;
+		String path = f->get_line().strip_edges();
```

Comment:
Maybe add a comment that this is sort of brittle since it requires the fields to appear in a specific order.